examples/jqlike/src/main/scala/fs2/data/example/jqlike/JqLike.scala

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2023 Lucas Satabin
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fs2.data.example.jqlike
+
+import cats.effect.{ExitCode, IO}
+import cats.syntax.all._
+import com.monovore.decline.Opts
+import com.monovore.decline.effect.CommandIOApp
+import fs2.Stream
+import fs2.data.json.jq.{Jq, JqParser}
+import fs2.data.json.{jq, render, tokens}
+import fs2.io.file.{Files, Path}
+
+object JqLike extends CommandIOApp(name = "fs2-jq", header = "A streaming implementation of a jq-like tool") {
+
+  val query: Opts[Option[String]] =
+    Opts
+      .option[String](long = "query", short = "q", help = "The query to execute on the input (default to '.')")
+      .orNone
+
+  val input: Opts[Either[String, Path]] =
+    Opts
+      .option[String](long = "input", short = "i", help = "The input json string")
```

Comment:
Maybe read `stdin` instead? The only `jq` use cases I ever had involved piping JSON into it…

Reply:
For simplicity I made the choice to have everything as explicit option and not positional arguments. So this prevents simple piping. This is an example, not a fully fledged json tool.